packages/gatsby/src/utils/webpack-utils.js

```diff
@@ -326,20 +341,20 @@ module.exports = async ({
    */
   {
     let dependencies = (
-      { modulesThatUseGatsby, ...options } = { modulesThatUseGatsby: [] }
+      { modulesThatUseGatsby } = { modulesThatUseGatsby: [] }
     ) => {
       const jsOptions = {
         babelrc: false,
         configFile: false,
-        compact: false,
-        presets: [
-          [require.resolve(`babel-preset-gatsby/dependencies`), { stage }],
-        ],
+        presets: [require.resolve(`babel-preset-gatsby/dependencies`)],
         // If an error happens in a package, it's possible to be
         // because it was compiled. Thus, we don't want the browser
         // debugger to show the original code. Instead, the code
         // being evaluated would be much more helpful.
         sourceMaps: false,
+        cacheIdentifier: `gatsby-dependencies@${
```

Comment:
`cacheIdentifier` by default also includes NODE_ENV or BABEL_ENV since that can affect builds. Do we need to include it here? 

https://github.com/babel/babel-loader/pull/110/files

Reply:
Normally yes but we don't run this outside production. I could add it just to be safe? WDYT?

Reply:
I would add it to guard it against changes in the future that might miss this 🤷🏻‍♂️

Reply:
Updated :)